### ISSUE TITLE
blur not working on app

### ISSUE THREAD

**Issue by kaushiksahu18** [aunetx/blur-my-shell#577]
when i first open an application blur will not work but if i switch window (slide forword and then backword) it works.
here is video explation
***
[Screencast from 2024-04-19 23-38-11.webm](https://github.com/aunetx/blur-my-shell/assets/113049724/91c48965-1d36-4011-aa63-94e5ec38386f)
***

there are only some apps that showing this behaviour like brave, vscode, warp etc...
> i use
> gnome 46

**Comment by aunetx** [comment]
Does something change when you disable "opaque focused window"? What you have sounds like a bug that could be due to that

Make sure to disable / re-enable the extension after turning off this option to be sure

**Comment by kaushiksahu18** [comment]
> Does something change when you disable "opaque focused window"? What you have sounds like a bug that could be due to that

@aunetx i don't know what this option is, and where i can find it (i am newbie)
**I will try searching in internet but your help will be best!**

**Comment by aunetx** [comment]
No problem! In blur-my-shell's preferences, in the applications tab, there should be an option "Opaque focused window" (or similar, maybe in your own language if your computer is not in English). If it is enabled, then the window that you currently use does not have blur (to help legibility and increase performances), but that may be linked to your issue.

**Comment by kaushiksahu18** [comment]
> If it is enabled, then the window that you currently use does not have blur

Yes now i remenber, after new update, i have saw that new feture (actualy i did't like that), so i have't enable that option for my desktop,
"So that is not the problem in my case", any other problem i guess
***
> but if i switch window (slide forword and then backword) it works.
**Then it wont be working after switch, i guess**

**Comment by FelixFourcolor** [comment]
I have the same problem. "Opaque focused window" is disabled for me. Some apps always start up opaque and only start blurring when you switch focus.

> there are only some apps that showing this behaviour like brave, vscode, warp etc...

I believe the common factor is XWayland.

**Comment by deepinsideDrury** [comment]
[录屏 2024-04-24 01-31-44.webm](https://github.com/aunetx/blur-my-shell/assets/26105172/495cfe47-4883-423e-bca3-1df1ed54d483)

**Comment by FelixFourcolor** [comment]
To reiterate that this is not OP's misunderstanding the setting "opaque focused window", but is in fact a bug, here's a video demonstration. The setting is off the entire time.

This video demonstrates the behavior with steam and vscode. When a window is first opened, it's always opaque, and only starts blurring when focus changes, and then remains blurred thereafter (even when focus changes back) until closed.

[Screencast from 2024-04-25 22-56-15.webm](https://github.com/aunetx/blur-my-shell/assets/34260584/ecaba397-40c1-443f-aad6-6fd0db0f2f0d)

(I have focus follows mouse.)
 
 > when i first open an application blur will not work but if i switch window (slide forword and then backword) it works
 
OP's original description agrees with mine. Their video also demonstrates the same problem as mine. By switching workspace back and forth, they effectively switch focus back and forth.

However this only happens with a few apps. For me it's steam and vscode. OP reported it also happens with brave and warp. I believe XWayland is the common factor in all of those apps.

**Comment by FelixFourcolor** [comment]
@aunetx Any updates on this?

@kaushiksahu18 If you agree with my speculation that this has something to do with XWayland, perhaps you could rename the title to something like "application blur issue with XWayland". It may make it easier for others to discover this thread.

**Comment by kaushiksahu18** [comment]
> @aunetx Any updates on this?
yes its getting irritating, everytime i have to do focus change-unchange!

> "application blur issue with XWayland"
then suggest more relatable title, i.g. add 'not work first time but work if do focus change-unchange' or similar you know. 👉

**Comment by stoneC0der** [comment]
> Does something change when you disable "opaque focused window"? What you have sounds like a bug that could be due to that
> 
> Make sure to disable / re-enable the extension after turning off this option to be sure

This is working on my side.

**Comment by FelixFourcolor** [comment]
> > Does something change when you disable "opaque focused window"? What you have sounds like a bug that could be due to that
> > Make sure to disable / re-enable the extension after turning off this option to be sure
> 
> This is working on my side.

It doesn't work for me. What apps did you test? This is an XWayland-specific issue, to see it you need to run Wayland and test X11 apps such as steam, vscode, discord, etc.

**Comment by stoneC0der** [comment]
> > > Does something change when you disable "opaque focused window"? What you have sounds like a bug that could be due to that
> > > Make sure to disable / re-enable the extension after turning off this option to be sure
> > 
> > 
> > This is working on my side.
> 
> It doesn't work for me. What apps did you test? This is an XWayland-specific issue, to see it you need to run Wayland and test X11 apps such as steam, vscode, discord, etc.

I am using X11, I have Wayland disabled by default since I set Nvidia as my primary GPU.
However, when I enable the option, the window in focus is not blurred, but the one in background is blurred, that include vscode.

**Comment by FelixFourcolor** [comment]
@stoneC0der What you described is not an issue, that's just how the "opaque focused window" setting works. What I and @kaushiksahu18 are referring to is a specific issue when that setting is *already disabled*. I can only reproduce it in wayland, so you don't have to worry about it.

I'm disappointed that this thread has attracted so much misunderstanding. Apparently only OP and I are on the same page about this. For future folks coming to this thread: **if your blur issue is fixed by turning off "opaque focused window", that's not the issue we're talking about**.

**Comment by kaushiksahu18** [comment]
> **if your blur issue is fixed by turning off "opaque focused window", that's not the issue we're talking about**.

### Yes because I don't even use/ON that "opaque focused window" option, and still facing issue!

**Comment by james-math** [comment]
I have the exact same issue when using Kitty (when the app first launches it has no blur effect and blur only begins after switching focus to another app at which point blur remains even when focus comes back).  I've got "linux_display_server x11" set in my kitty.conf file and when I removed this line the issue disappeared which seems to confirm that this is an issue with XWayland specifically.

**Comment by kaushiksahu18** [comment]
> application blur issue with XWayland

here it is,.

**Comment by FelixFourcolor** [comment]
@aunetx Can you take a look back at this old issue? Last time you came there were some confusion about what the issue is, but we in the comments have reached a consesus that it has something to do with Xwayland.
